File: trunk/src/core/src/main/java/org/semachina/jena/features/Feature.java
package org.semachina.jena.features;

import org.semachina.jena.SemachinaOntModel;

/**
 * Created by IntelliJ IDEA.
 * User: sgopalan
 * Date: Dec 8, 2010
 * Time: 8:34:38 AM
 * To change this template use File | Settings | File Templates.
 */
public interface Feature {

    String getKey();
    
    void init(SemachinaOntModel ontModel, org.semachina.jena.SemachinaFactory factory) throws Exception;

    void close() throws Exception;
}

File: trunk/src/core/src/main/scala/org/semachina/jena/config/features/Feature.scala
package org.semachina.jena.config.features

import com.hp.hpl.jena.ontology.OntModel

/**
 * Created by IntelliJ IDEA.
 * User: sgopalan
 * Date: Dec 8, 2010
 * Time: 8:34:38 AM
 * To change this template use File | Settings | File Templates.
 */
trait Feature {
  val key: String

  def getKey() : String = {
    return key
  }

  def init(ontModel: OntModel): Unit

  def close: Unit
}